src/main/java/org/medicmobile/webapp/mobile/SettingsDialogActivity.java

```diff
@@ -109,7 +80,23 @@ public void verifyAndSave(View view) {
 
 		String appUrl = text(R.id.txtAppUrl);
 
-		new AppUrlVerificationTask(this).execute(appUrl);
+		TaskRunner taskRunner = new TaskRunner();
+		taskRunner.executeAsync(new AppUrlVerificationTask(appUrl), (result) -> {
+			trace(
+				this,
+				"SettingsDialogActivity :: Executing verification callback, result isOkay=%s, appUrl=%s",
+				result.isOk, result.appUrl
+			);
+
+			if (result.isOk) {
```

Comment:
This is from [here](https://github.com/medic/cht-android/blob/ebe095c6432d707e5b153c1526e4b82eded2f77a/src/main/java/org/medicmobile/webapp/mobile/SettingsDialogActivity.java#L48).